### PR TITLE
[ADD] New Module mrp_production_hierarchy

### PR DESCRIPTION
**Pull request by andrea4ever**
Add a hierarchy for class mrp.production and show in a tree like Product for Categoryx:

### CODE REVIEW

File: mrp_production_hierarchy/README.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+========================
+Mrp Production Hierarchy
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: s/Mrp/MRP

File: mrp_production_hierarchy/README.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+========================
+
+This module was written to extend the functionality of Manufacturing Order
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: This module extends the functionality of Manufacturing Order

File: mrp_production_hierarchy/README.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+to show a navigable hierarchy
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: period

File: mrp_production_hierarchy/README.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Installation
+============
+
+To install this module, you need to:
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: No installation steps required

File: mrp_production_hierarchy/README.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+* This module depends only on the mrp module.
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: remove

File: mrp_production_hierarchy/README.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+Configuration
+=============
+
+To configure this module, you need to:
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: No configuration steps required

File: mrp_production_hierarchy/README.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+* No special setup
+
+Usage
+=====
+
+To use this module, you need to:
+
+* go to menù Manufacturing - Manufacturing - Production Hierarchy
```
Comment (elicoidal) [inline mrp_production_hierarchy/README.rst]: s/menù/menu

File: mrp_production_hierarchy/__init__.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline mrp_production_hierarchy/__init__.py:1]: simplify headers following: https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3
(several)

File: mrp_production_hierarchy/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Mrp Production Hierarchy',
+    'version': '8.0.1.0.0',
+    'category': 'mrp',
+    'author': 'Apulia Software srl, Odoo Community Association (OCA)',
```
Comment (elicoidal) [inline mrp_production_hierarchy/__openerp__.py:12]: remove extra space

Comment (eLBati) [inline mrp_production_hierarchy/__openerp__.py:12]: Hello @elicoidal if you talk about this `srl, Odoo` , I don't think the removal is actually needed. See https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L10 indeed

File: mrp_production_hierarchy/.gitignore [hunk 10]
```diff
@@ -0,0 +1,108 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (eLBati) [inline mrp_production_hierarchy/.gitignore]: Please remove this file

File: mrp_production_hierarchy/__openerp__.py (hunk 9 above)
Comment (elicoidal) [inline mrp_production_hierarchy/__openerp__.py:12]: @eLBati You are right!

File: mrp_production_hierarchy/README.rst [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/129/8.0
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
+{project_repo}/issues>`_.
```
Comment (eLBati) [inline mrp_production_hierarchy/README.rst]: Replace {project_repo}

File: mrp_production_hierarchy/README.rst [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback `here <https://github.com/OCA/
+{project_repo}/issues/new?body=module:%20
+{module_name}%0Aversion:%20
+{branch}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (eLBati) [inline mrp_production_hierarchy/README.rst]: Reaplce {project_repo} {module_name} and {branch}

File: mrp_production_hierarchy/README.rst [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+Usage
+=====
+
+To use this module, you need to:
+
+* go to menù Manufacturing - Manufacturing - Production Hierarchy
+
+or
+
+* Select from a selected Manufactoring Order Wizard Production Hierarchy
```
Comment (eLBati) [inline mrp_production_hierarchy/README.rst]: If I understand what you mean, I would say `Run Production Hierarchy wizard from the production order form`

File: mrp_production_hierarchy/models/mrp.py [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+from openerp import models, fields, api
+
+
+class MrpProduction(models.Model):
+
+    _inherit = 'mrp.production'
+
+    @api.one
+    @api.depends('name', 'state')
```
Comment (eLBati) [inline mrp_production_hierarchy/models/mrp.py]: This should also contain `origin` field

**Comment by andrea4ever** [comment]
Comment done! Thank's for your review! @eLBati @elicoidal

**Comment by eLBati** [comment]
I noticed an issue with this module: when production orders are a lot, `Production Hierarchy` view can be heavy to load by the browser (as the list is not paged).
Maybe this deserves a comment in README?

**Comment by andrea4ever** [comment]
@eLBati add comment in README for browser long time loading

**Comment by scigghia** [comment]
:+1:

**Comment by LoisRForgeFlow** [comment]
Closing as this is stale. Please reopen if you think still is relevant.

**Pull request closed without merging**